### ISSUE TITLE
REPL over UART not working on the RP2 port

### ISSUE THREAD

**Issue by Sebi43217** [micropython/micropython#11057]
I’m trying to run a clean micropython port for the rp2 using a picoprobe as a debugger but I’m unable to use the REPL over UART functionality. 

* The only modification from a clean micropiton checkout is that i enabled  MICROPY_HW_ENABLE_UART_REPL in the mpconfigport.h

* The REPL over UART sems to immediately crash after startup since its not responding after it has send its startup information.

* When using rshell its getting stuck at "Retrieving sysname ..."
<img width="527" alt="image" src="https://user-images.githubusercontent.com/43995448/225936708-b42afc37-2a1f-40c3-9c28-a0e24cea6c1a.png">

* Using a terminal like minicom I can see the startup message but I’m unable to get any future response from the rp2.
<img width="545" alt="image" src="https://user-images.githubusercontent.com/43995448/225937347-91add04a-aa3f-476e-a506-162e03c93ac9.png">

**Comment by robert-hh** [comment]
I can enable UART REPL and it works fine, with and without USB CDC enabled. Tested with a simple USB/UART adapter.

**Comment by robert-hh** [comment]
I just noticed that UART REPL input only works if DEBUG is disabled. For builds with DEBUG=1 uart REPL does not take input values.

**Comment by robert-hh** [comment]
UART0 seems to be disabled by DEBUG code in the SDK. But there is a hack in the code to re-enable it.
Add to main.py the lines:
```
from machine import freq
freq(125000000)
```
The actual freq value does not matter, but the default is fine.

**Comment by Sebi43217** [comment]
Thanks that could be the solution since I was always working with DEBUG enabled.

Sadly I'm away for the weekend but I'll try it out on Monday.

**Comment by robert-hh** [comment]
I made a branch with a suitable change. There is one line changed in main.c, line 93-95, to:
```
    #if !defined(NDEBUG) && !MICROPY_HW_ENABLE_UART_REPL
    stdio_init_all();
    #endif
```
Actually, only line 93 is changed. And I removed the patch in modmachine.c in the code for machine.freq().
The branch is here: https://github.com/robert-hh/micropython/tree/rp2_keep_uart0

**Comment by dpgeorge** [comment]
Fixed by 711bac511e547096fe1e58af3ba70d31b244731f
